pkg/deployments/tasks/20220415-primary-issue-pool/test/test.fork.ts

```diff
@@ -86,27 +108,150 @@ describeForkTest('PrimaryPoolFactory', 'goerli', 8586768, function () {
 
             const { balances } = await vault.getPoolTokens(poolId);
             expect(balances).to.deep.equal(initialBalances);
-
         });
 
+
         it('swap in the pool', async () => {
-            const amount = fp(500);
-            await vcusd.connect(whale).transfer(owner.address, amount);
-            await vcusd.connect(owner).approve(vault.address, amount);
-
-            await vault
-                .connect(owner)
-                .swap(
-                { kind: SwapKind.GivenIn, poolId, assetIn: VCUSD, assetOut: USDC, amount, userData: '0x' },
-                { sender: owner.address, recipient: owner.address, fromInternalBalance: false, toInternalBalance: false },
-                0,
-                MAX_UINT256
-                );
-
-            // Assert pool swap
-            const expectedUSDC = amount.div(1e12);
-            expectEqualWithError(await vcusd.balanceOf(owner.address), 0, 0.0001);
-            expectEqualWithError(await usdc.balanceOf(owner.address), expectedUSDC, 0.1);
+            try {
+                const amount = fp(500);
+
+
+                // const balance = await vcusd.balanceOf(owner.address);
+                // console.log("Owner's VCUSD balance: ", balance.toString());
+
+                // if (balance.lt(amount)) {
+                //     throw new Error('Insufficient VCUSD balance for the swap');
+                // }
+
+                // if ((await vcusd.balanceOf(owner.address)).lt(amount)) {
+                //     throw new Error('Insufficient VCUSD balance for the swap');
+                // }
+
+
+                await usdc.connect(whale).transfer(owner.address, amount);
+                await usdc.connect(owner).approve(vault.address, amount);
+
+                await vault
+                    .connect(owner)
+                    .swap(
+                        { kind: SwapKind.GivenIn, poolId, assetIn: usdc, assetOut: usdc, amount, userData: '0x' },
```

Comment:
@vinaykumar0103 swap assetIn and assetOut are the same ? They should be different. And 'amount' of assets swapped in should be there in balance.